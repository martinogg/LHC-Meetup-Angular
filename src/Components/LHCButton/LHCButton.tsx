
import React, { Component, Children } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

interface IProps {
    onSelected(): void
    styles?: any
}

export default class LHCButton extends Component<IProps> {

    constructor(props: IProps) {

        super(props);
    }

    render() {

        return <TouchableOpacity style={[localStyles.container, this.props.styles]} onPress={() => this.props.onSelected()}>
            {this.props.children}
        </TouchableOpacity>
    }
}

const localStyles = StyleSheet.create({
    container: {
        margin: 10,
        backgroundColor: '#2980b6',
        paddingVertical: 15
    },
})