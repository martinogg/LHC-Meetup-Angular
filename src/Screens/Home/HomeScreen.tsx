import React, { Component } from 'react';
import { Button, Platform, StyleSheet, Text, View } from 'react-native';
import { AppStyles } from '../../AppStyles'

const instructions = Platform.select({
    ios: 'Press Cmd+R to reload,\n' + 'Cmd+D or shake for dev menu',
    android:
        'Double tap R on your keyboard to reload,\n' +
        'Shake or press menu button for dev menu',
});

interface Props { navigation: any }

export class HomeScreen extends Component<Props> {
    private static navigationOptions = {
        title: 'Welcome',
    };
    public render() {
        const { navigate } = this.props.navigation;
        return (
            <View style={AppStyles.container}>
                <Text style={styles.welcome}>Welcome to HOME!</Text>
                <Text style={styles.instructions}>To get started, edit App.tsx</Text>
                <Text style={styles.instructions}>{instructions}</Text>
                <Button title='About' 
                    onPress={() => navigate('AboutApp')}
                />
                <Button title='Edit' 
                    onPress={() => navigate('EditDetails')}
                />
                <Button title='Browse' 
                    onPress={() => navigate('Browse')}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    welcome: {
        fontSize: 20,
        textAlign: 'center',
        margin: 10,
    },
    instructions: {
        textAlign: 'center',
        color: '#333333',
        marginBottom: 5,
    },
});
