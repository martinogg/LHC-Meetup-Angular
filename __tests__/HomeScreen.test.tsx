import { shallow } from 'enzyme';
import React from 'react';
import 'react-native';
import LHCButton from '../src/Components/LHCButton/LHCButton'

import { HomeScreen } from '../src/Screens/Home/HomeScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('should display HomeScreen with no errors', () => {

  const navigation = { navigate: jest.fn() };
  expect(renderer.create(<HomeScreen navigation={navigation} />)).toMatchSnapshot();
});

test('test onPress About functionality', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<HomeScreen navigation={navigation} />);

  wrapper.find(LHCButton).at(2).simulate('selected')

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('AboutApp')
});

test('test onPress Profile functionality', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<HomeScreen navigation={navigation} />);

  wrapper.find(LHCButton).first().simulate('selected')

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('Profile', {editable: true})
});

test('test onPress Browse functionality', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<HomeScreen navigation={navigation} />);

  wrapper.find(LHCButton).at(1).simulate('selected')

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('Browse')
});