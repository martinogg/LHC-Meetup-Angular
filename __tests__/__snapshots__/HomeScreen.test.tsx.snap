// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display HomeScreen with no errors 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#2c3e50",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontWeight": "700",
          "textAlign": "center",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    Hello! Create your profile to Begin, then Search for other people
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b6",
        "margin": 10,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Enter Profile Details
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b6",
        "margin": 10,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Browse by Interest
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b6",
        "margin": 10,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Invitations
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2980b6",
        "margin": 10,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      About this App
    </Text>
  </View>
</RCTSafeAreaView>
`;
